docs/compilers/CSharp/Compiler Breaking Changes - DotNet 10.md

```diff
@@ -0,0 +1,43 @@
+# This document lists known breaking changes in Roslyn after .NET 8 all the way to .NET 9.
+
+## `Span` and `ReadOnlySpan` overloads are applicable in more scenarios in C# 14 and newer
+
+***Introduced in Visual Studio 2022 version 17.13***
+
+C# 14 introduces new [built-in span conversions and type inference rules](https://github.com/dotnet/csharplang/issues/7905).
+This means that different overloads might be chosen compared to C# 13, and sometimes an ambiguity compile-time error
+might be raised because a new overload is applicable but there is no single best overload.
+
+The following example shows some ambiguities and possible workarounds.
+Note that another workaround is for API authors to use `OverloadResolutionPriorityAttribute`.
+
+```cs
+var x = new long[] { 1 };
+Assert.Equal([2], x); // previously Assert.Equal<T>(T[], T[]), now ambiguous with Assert.Equal<T>(ReadOnlySpan<T>, Span<T>)
+Assert.Equal([2], x.AsSpan()); // workaround
+
+var y = new int[] { 1, 2 };
+var s = new ArraySegment<int>(x, 1, 1);
+Assert.Equal(y, s); // previously Assert.Equal<T>(T, T), now ambiguous with Assert.Equal<T>(Span<T>, Span<T>)
+Assert.Equal(y.AsSpan(), s); // workaround
+```
+
+A `Span` overload might be chosen in C# 14 where an `IEnumerable` overload was chosen in C# 13,
+and that can lead to `ArrayTypeMismatchException`s at runtime if you are using covariant arrays:
```

Comment:
>`ArrayTypeMismatchException`s at runtime if you are using covariant arrays [](http://example.com/codeflow?start=21&length=74)

Minor: Perhaps avoid the plural with "an `ArrayTypeMismatchException` at runtime if the array is covariant"